pkg/types/gcp/validation/platform.go

```diff
@@ -65,8 +67,34 @@ var (
 		sort.Strings(validValues)
 		return validValues
 	}()
+
+	// userLabelKeyRegex is for verifying that the label key contains only allowed characters.
+	userLabelKeyRegex = regexp.MustCompile(`^[a-z][0-9a-z_-]{1,62}$`)
+
+	// userLabelValueRegex is for verifying that the label value contains only allowed characters.
+	userLabelValueRegex = regexp.MustCompile(`^[0-9a-z_-]{1,63}$`)
+
+	// userLabelKeyPrefixRegex is for verifying that the label key does not contain restricted prefixes.
+	userLabelKeyPrefixRegex = regexp.MustCompile(`^(?i)(kubernetes\-io|openshift\-io)`)
+
+	// userTagKeyRegex is for verifying that the tag key contains only allowed characters.
+	userTagKeyRegex = regexp.MustCompile(`^[a-zA-Z0-9]([0-9A-Za-z_.-]{0,61}[a-zA-Z0-9])?$`)
+
+	// userTagValueRegex is for verifying that the tag value contains only allowed characters.
+	userTagValueRegex = regexp.MustCompile(`^[a-zA-Z0-9]([0-9A-Za-z_.@%=+:,*#&()\[\]{}\-\s]{0,61}[a-zA-Z0-9])?$`)
+
+	// userTagParentIDRegex is for verifying that the tag parentID contains only allowed characters.
+	userTagParentIDRegex = regexp.MustCompile(`(^[1-9][0-9]{0,31}$)|(^[a-z][a-z0-9-]{4,28}[a-z0-9]$)`)
```

Comment:
As user tags are pre-existing resources, can we validate the user input by passing it directly to a GCP API call to get the user tag? Unless these regexes are documented/supported by GCP, I think they provide more risk than benefit. 
I'm not sure if it is present in a separate PR, but I think we need validation to ensure the specified user tags indeed exist.

Reply:
Explicit check on the tag key or tag value existence is not done, but the code [here](https://github.com/openshift/installer/pull/7279/files#diff-fe206eadc78bb637c8b5ebbf96a7fe49eb5eab26b5ce28153f793d8d4c1f72a7R175) fails if the either of them doesn't exist.

Regex created based on the requirements defined by GCP for tag keys and values, same as labels. I think this will be helpful in catching any errors with tags saving the API invocations. Please let me know WDYT?

Reply:
If you want to leave the regexes in, that works for me. I'm not a regex expert but I looked over them carefully and they look good. My preference/advice is still as originally stated, but as far as I can tell this is fine.